include/mega/types.h

```diff
@@ -262,6 +263,27 @@ typedef enum { NOTLOGGEDIN, EPHEMERALACCOUNT, CONFIRMEDACCOUNT, FULLACCOUNT } se
 // in a 64-bit int
 typedef uint64_t handle;
 
+class NodeHandle
+{
+    // Handles of nodes are only 6 bytes.
+    // This class helps avoid issues when we don't save/restore the top 2 bytes when using an 8 byte uint64 to represent it
+    uint64_t h = 0xFFFFFFFFFFFFFFFF;
+public:
+    bool isUndef() { return (h & 0xFFFFFFFFFFFF) == 0xFFFFFFFFFFFF; }
+    NodeHandle& set6byte(uint64_t n) { h = n; assert((n & 0xFFFF000000000000) == 0); return *this; }
```

Comment:
Perhaps it would be nice to define some masks, for easier reading (I've already needed to count the `F`s many times).
Something like:
``` cpp
static const uint64_t leastSignificant6B = 0x0000FFFFFFFFFFFF;
static const uint64_t mostSignificant2B = 0xFFFF000000000000;
```

Reply:
I was just aiming to keep the class small (ie fewer lines) ; the main thing is that the interface lets us use NodeHandles simply and effectively in the rest of the code.   As to whether 1+3 is clearer than 3+1, well, maybe :-)